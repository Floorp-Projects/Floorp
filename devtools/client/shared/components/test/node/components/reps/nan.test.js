/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

"use strict";

const { shallow } = require("enzyme");

const {
  REPS,
  getRep,
} = require("devtools/client/shared/components/reps/reps/rep");

const { NaNRep, Rep } = REPS;

const stubs = require("devtools/client/shared/components/test/node/stubs/reps/nan");

describe("NaN", () => {
  const stub = stubs.get("NaN");

  it("selects NaN Rep as expected", () => {
    expect(getRep(stub)).toBe(NaNRep.rep);
  });

  it("renders NaN Rep as expected", () => {
    const renderedComponent = shallow(
      Rep({
        object: stub,
      })
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it("NaN rep renders with the correct title element", () => {
    const renderedComponent = shallow(
      Rep({
        object: stub,
        shouldRenderTooltip: true,
      })
    );
    expect(renderedComponent.prop("title")).toBe("NaN");
  });
});
