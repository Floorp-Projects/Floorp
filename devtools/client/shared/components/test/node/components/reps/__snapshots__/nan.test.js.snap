// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NaN renders NaN Rep as expected 1`] = `
<span
  className="objectBox objectBox-nan"
  title={null}
>
  NaN
</span>
`;
